packages/web3-providers-ws/src/index.ts

```diff
@@ -0,0 +1,391 @@
+				undefined,
+				this.wsProviderOptions?.headers,
+				this.wsProviderOptions?.requestOptions,
+				this.wsProviderOptions?.clientConfig,
+			);
+
+			this.addSocketListeners();
+		} catch (e) {
+			this.providerStatus = 'disconnected';
+			throw new InvalidConnectionError(this.clientUrl);
+		}
+	}
+
+	public disconnect(code: number, reason: string): void {
+		this.removeSocketListeners();
+		this.webSocketConnection?.close(code || 1000, reason);
+		this.providerStatus = 'disconnected';
+	}
+
+	public reset(): void {
+		this.sentQueue.clear();
+		this.requestQueue.clear();
+
+		this.init();
+		this.removeSocketListeners();
+		this.addSocketListeners();
+	}
+
+	public async request<T = JsonRpcResponse, T2 = unknown[]>(
+		request: JsonRpcPayload<T2>,
+	): Promise<T> {
+		if (this.webSocketConnection === undefined)
+			throw new Error('WebSocket connection is undefined');
+
+		if (request.id === undefined) throw new Error('Request Id not defined');
+
+		if (this.requestQueue.has(request.id) || this.sentQueue.has(request.id))
+			throw new Error(
+				'Duplicate request Id. Another request is already in Queue with same request Id.',
+			);
+
+		const { id } = request;
+
+		if (
+			this.webSocketConnection.readyState === this.webSocketConnection.CLOSED ||
+			this.webSocketConnection.readyState === this.webSocketConnection.CLOSING
+		) {
+			this.providerStatus = 'disconnected';
+
+			this.requestQueue.delete(id);
+
+			throw new ConnectionNotOpenError();
+		}
+
+		const requestItem = this.requestQueue.get(id);
+		if (this.webSocketConnection.readyState === this.webSocketConnection.CONNECTING) {
+			this.providerStatus = 'connecting';
+
+			if (requestItem === undefined) {
+				const defPromise = new DeferredPromise<T>();
+
+				const reqItem: WSRequestItem<T2, T> = {
+					payload: request,
+					deferredPromise: defPromise,
+				};
+
+				this.requestQueue.set(id, reqItem);
+				return defPromise.realPromise;
+			}
+
+			return requestItem.deferredPromise.realPromise as Promise<T>;
+		}
+
+		let promise;
+
+		if (requestItem !== undefined) {
+			this.sentQueue.set(id, requestItem);
+			this.requestQueue.delete(id);
+			promise = requestItem.deferredPromise.realPromise as Promise<T>;
+		} else {
+			const defPromise = new DeferredPromise<T>();
+
+			const reqItem: WSRequestItem<T2, T> = {
+				payload: request,
+				deferredPromise: defPromise,
+			};
+
+			this.sentQueue.set(id, reqItem);
+			promise = defPromise.realPromise;
+		}
+
+		try {
+			this.webSocketConnection.send(JSON.stringify(request));
+		} catch (error) {
+			this.sentQueue.delete(id);
+			throw error;
+		}
+
+		return promise;
+	}
+
+	public removeAllListeners(type: string): void {
+		this.wsEventEmitter.removeAllListeners(type);
+	}
+
+	private init() {
+		this.lastDataChunk = '';
+		this.reconnectAttempts = 0;
+
+		if (this.lastChunkTimeout) {
+			clearTimeout(this.lastChunkTimeout);
+		}
+	}
+
+	private addSocketListeners(): void {
+		// eslint-disable-next-line
```

Comment:
I think an explanation (like error message +  reasons why it's not being fixed) should be provided when disabling linting. Additionally, you should be able to disable the specific rule for the line instead of disabling linting entirely